### PR TITLE
Fix failure to dispose stream when exception occurs in Package.Open 

### LINKED ISSUE #46791: File not released when exception occurs in Package.Open
### Description

When I try to call `Package.Open` on an invalid package (file attached), the file lock is not released. The subsequent `File.Move` throws an `IOException`: `The process cannot access the file because it is being used by another process`

```c#
using System.IO;
using System.IO.Packaging;

internal class Program
{
    private static void Main(string[] args)
    {
        try
        {
            using var package = Package.Open("PdfWithFakeExtension.xlsx", FileMode.Open, FileAccess.Read, FileShare.Read);
        }
        catch
        {
            // Swallow exception
        }

        File.Move("PdfWithFakeExtension.xlsx", @"C:\temp\PdfWithFakeExtension.xlsx");
    }
}

```
### Configuration

* Which version of .NET is the code running on?
`netcore3.1`

* What OS and version, and what distro if applicable?
`System.IO.Packaging` version 5.0.0
* What is the architecture (x64, x86, ARM, ARM64)?
`AnyCPU`

### Regression?

Yes, I think this is a regression. Using a `net472` build and using `WindowsBase` instead of the `System.IO.Packaging` NuGet package, the file is released, so I don't experience this problem.

### Other information

[PdfWithFakeExtension.xlsx](https://github.com/dotnet/runtime/files/5792595/PdfWithFakeExtension.xlsx)

### PR DESCRIPTION
**Pull request by danmoseley**
Fix https://github.com/dotnet/runtime/issues/46791

Note these tests run against .NET Framework as well, demonstrating indeed it is a regression. (On .NET Framework, the garbling produces ArgumentOutOfRangeException, but it does dispose the stream.)

This test will pass even with the bug if the finalizer thread is lucky, but in my tests it consistently fails without the fix.

### CODE REVIEW

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs [hunk 1]
```diff
@@ -287,6 +287,11 @@ internal ZipPackage(string path, FileMode packageFileMode, FileAccess packageFil
                     zipArchive.Dispose();
                 }
 
+                if (_containerStream != null)
+                {
+                    _containerStream.Dispose();
+                }
+
```
Comment (stephentoub) [inline src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs:294]: Nit:
```C#$
zipArchive?.Dispose();
_containerStream?.Dispose();
```

File: src/libraries/System.IO.Packaging/tests/Tests.cs [hunk 2]
```diff
@@ -181,6 +181,27 @@ public void PackageOpen_Open_EmptyStream_Throws()
             }
         }
 
+        [Fact]
+        public void PackageOpen_Open_InvalidContent_Throws()
+        {
+            var temp = GetTempFileInfoWithExtension(".docx").FullName;
```
Comment (stephentoub) [inline src/libraries/System.IO.Packaging/tests/Tests.cs:187]:
```suggestion
            string temp = GetTempFileInfoWithExtension(".docx").FullName;
```

File: src/libraries/System.IO.Packaging/tests/Tests.cs [hunk 3]
```diff
@@ -181,6 +181,27 @@ public void PackageOpen_Open_EmptyStream_Throws()
...
+
+            using (FileStream fs = File.OpenWrite(temp))
+            {
+                byte[] bytes = File.ReadAllBytes("plain.docx");
+                fs.Write(bytes, 0, bytes.Length);
+                fs.Seek(500, SeekOrigin.Begin);
+                fs.Write(new byte[500], 0, 500); // garble it
```
Comment (stephentoub) [inline src/libraries/System.IO.Packaging/tests/Tests.cs:194]:
```suggestion
                byte[] bytes = File.ReadAllBytes("plain.docx");
                bytes.Slice(500, 500).Clear();
                fs.Write(bytes, 0, bytes.Length);
```
?

Comment (danmosemsft) [reply src/libraries/System.IO.Packaging/tests/Tests.cs:194]: Yes, neater. I forgot we have spans on .NET Framework.

**Comment by danmosemsft** [comment]
@stephentoub any other feedback?

**Review by stephentoub — APPROVED** [review]

**Comment by danmosemsft** [comment]
@dotnet/dnceng , the tests for release-coreclr-windows-x64-debug have not completed for this particular queue below after over 90 minutes. When I look at the helix status, not a single one of the jobs has completed -- they're all marked Waiting. Is the queue backed up?
```
  Sending Job to (Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64-08e8e40-20200107182504...
  Sent Helix Job; see work items at https://helix.dot.net/api/jobs/c745d70a-4e1f-4ea1-b04b-6a89b83ffea7/workitems?api-version=2019-06-17
```

**Comment by Chrisboh** [comment]
@AlitzelMendez or @adiaaida could one of you take a look and see what is going on?

**Comment by adiaaida** [comment]
Hey @danmosemsft, looks like something went wrong with the autoscaler. We've scaled in and out on this queue, and it is now taking work. @AlitzelMendez will take a look into what happened.

**Comment by danmosemsft** [comment]
Thanks both!

**Comment by danmosemsft** [comment]
cc @jaredpar fyi.

**Comment by AlitzelMendez** [comment]
As an update what happened in here was the same issue as https://github.com/dotnet/core-eng/issues/11444 .

The fix for this problem was  and rolled out today so this shouldn't happen again.

**Comment by danmosemsft** [comment]
thanks @AlitzelMendez

**Pull request merged**